x/incentives/keeper/distribute.go

```diff
@@ -343,33 +371,82 @@ func (k Keeper) Distribute(ctx sdk.Context, gauges []types.Gauge) (sdk.Coins, er
 	distrInfo := newDistributionInfo()
 
 	locksByDenomCache := make(map[string][]lockuptypes.PeriodLock)
-	totalDistributedCoins := sdk.Coins{}
+	totalDistributedCoins := sdk.NewCoins()
+
+	// get pool Id corresponding to the gaugeId
+	incentiveParams := k.GetParams(ctx).DistrEpochIdentifier
+	currentEpoch := k.ek.GetEpochInfo(ctx, incentiveParams)
+
 	for _, gauge := range gauges {
-		filteredLocks := k.getDistributeToBaseLocks(ctx, gauge, locksByDenomCache)
-		// send based on synthetic lockup coins if it's distributing to synthetic lockups
 		var gaugeDistributedCoins sdk.Coins
-		var err error
-		if lockuptypes.IsSyntheticDenom(gauge.DistributeTo.Denom) {
-			gaugeDistributedCoins, err = k.distributeSyntheticInternal(ctx, gauge, filteredLocks, &distrInfo)
+		clPool := k.GetValidConcentratedLiquidityGauge(ctx, gauge.Id, currentEpoch.Duration)
+		// only want to run this logic if the gaugeId is associated with CL PoolId
+		if clPool != nil {
```

Comment:
yea i can do that the only thing is we donot want to error in `GetValidConcentratedLiquidityGauge` because a gauge doesnot always have to be associated with a poolId